rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -261,7 +264,8 @@ rmw_destroy_client(rmw_node_t * node, rmw_client_t * client)
     }
     delete info;
   }
-  delete (client);
+  rmw_free(const_cast<char *>(client->service_name));
```

Comment:
> Since the rmw_free function just calls free, I think it's ok to remove the null check around the one above.

Yeah, in this instance, I agree with you, since free is guaranteed to do the correct thing on a NULL pointer.  However, if you think of rmw_free() as a "black box" function that you don't know anything about, then it is helpful to be conservative and not pass NULL into there.